server/migrations.js

```diff
@@ -1,26 +1,28 @@
-Migrations.add({
-  version: 1,
-  name: 'Move rounds.progress data to new roundProgresses collection.',
-  up: function() {
-    var allRounds = Rounds.find({}).fetch();
-    allRounds.forEach(function(round) {
-      var done = 0;
-      var total = 0;
-      if(round.progress) {
-        done = round.progress.done;
-        total = round.progress.total;
-      }
-      RoundProgresses.insert({
-        roundId: round._id,
-        competitionId: round.competitionId,
-        done: done,
-        total: total,
+Migrations.add(
+  {
+    version: 2,
+    name: 'Compute rounds.totalCount and remove rounds.roundCode.',
+    up: function() {
+      var counts = {};
+
+      Rounds.find({}).fetch().forEach(function(round) {
+        var key = round.competitionId + ',' + round.eventCode;
+        counts[key] = (counts[key] || 0) + 1;
+      });
+
+      Object.keys(counts).forEach(function(key) {
+        ids = key.split(',');
+        Rounds.update(
+          {competitionId: ids[0], eventCode: ids[1]},
+          {$set: {totalRounds: counts[key]}},
+          {multi: true}
+        );
       });
 
-      Rounds.update({_id: round._id}, { $unset: { progress: 1 } });
-    });
+      Rounds.update({}, {$unset: { roundCode: 1 }}, {multi: true});
```

Comment:
Hmm... when running this code myself, I saw the following exception:

```bash
I20150529-15:03:06.022(-7)? Migrating from version 0 -> 2
I20150529-15:03:06.032(-7)? Running up() on version 2 (Compute rounds.totalCount and remove rounds.roundCode.)
W20150529-15:03:06.039(-7)? (STDERR) 
W20150529-15:03:06.039(-7)? (STDERR) /home/jeremy/.meteor/packages/meteor-tool/.1.1.3.90rsz++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:245
W20150529-15:03:06.040(-7)? (STDERR) 						throw(ex);
W20150529-15:03:06.040(-7)? (STDERR) 						      ^
W20150529-15:03:06.078(-7)? (STDERR) Error: roundCode is not allowed by the schema
W20150529-15:03:06.078(-7)? (STDERR)     at getErrorObject (packages/aldeed:collection2/collection2.js:369:1)
W20150529-15:03:06.078(-7)? (STDERR)     at [object Object].doValidate (packages/aldeed:collection2/collection2.js:352:1)
W20150529-15:03:06.078(-7)? (STDERR)     at [object Object].Mongo.Collection.(anonymous function) [as update] (packages/aldeed:collection2/collection2.js:154:1)
W20150529-15:03:06.079(-7)? (STDERR)     at Migrations.add.up (app/server/migrations.js:22:14)
W20150529-15:03:06.079(-7)? (STDERR)     at migrate (packages/percolate:migrations/migrations_server.js:142:1)
W20150529-15:03:06.079(-7)? (STDERR)     at Object.Migrations._migrateTo (packages/percolate:migrations/migrations_server.js:153:1)
W20150529-15:03:06.079(-7)? (STDERR)     at Object.Migrations.migrateTo (packages/percolate:migrations/migrations_server.js:80:1)
W20150529-15:03:06.079(-7)? (STDERR)     at app/server/migrations.js:28:14
W20150529-15:03:06.079(-7)? (STDERR)     at /home/jeremy/tmp/ccm/.meteor/local/build/programs/server/boot.js:229:5
=> Exited with code: 8
```

To fix this, I passed [validate: false](https://github.com/aldeed/meteor-collection2#passing-options) to the update method:

`Rounds.update({}, {$unset: { roundCode: 1 }}, {multi: true, validate: false});`